From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Martin Blumenstingl <martin.blumenstingl@googlemail.com>
Date: Fri, 4 Jun 2021 21:50:06 +0200
Subject: ARM: dts: meson8: Add the shared CMA dma memory pool

The 4K HDMI modes needs more CMA memory (than the default 64MiB) to be
reserved at boot-time. Add a shared-dma-pool with increased size so the
4K HDMI modes can be used.

Signed-off-by: Martin Blumenstingl <martin.blumenstingl@googlemail.com>

[ported] The OneCloud's S805 is limited to 1.65GHz HDMI PHY (1080p), so
the pool is sized for 1080p framebuffers (64MiB) instead of the 256MiB
which this patch reserves for 4K modes. Note that meson8b.dtsi replaces
this /reserved-memory node, so on Meson8b the pool actually comes from
CONFIG_CMA_SIZE_MBYTES (also set to 64MiB); the value here is kept
consistent for Meson8/Meson8m2 boards.
---
 arch/arm/boot/dts/amlogic/meson8.dtsi | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/arch/arm/boot/dts/amlogic/meson8.dtsi b/arch/arm/boot/dts/amlogic/meson8.dtsi
index 111111111111..222222222222 100644
--- a/arch/arm/boot/dts/amlogic/meson8.dtsi
+++ b/arch/arm/boot/dts/amlogic/meson8.dtsi
@@ -193,6 +193,14 @@ power-firmware@4f00000 {
 			reg = <0x4f00000 0x100000>;
 			no-map;
 		};
+
+		linux,cma {
+			compatible = "shared-dma-pool";
+			reusable;
+			size = <0x4000000>;
+			alignment = <0x400000>;
+			linux,cma-default;
+		};
 	};
 
 	thermal-zones {
-- 
Armbian

